tst/com/amazon/ion/benchmark/DataGeneratorTest.java

```diff
@@ -117,57 +110,6 @@ public static void violationDetect(String inputFile) throws Exception {
         }
     }
 
-    /**
-     * Assert generated Ion data is the same type as expected.
-     * @throws Exception if error occurs when executing Ion data generator.
-     */
-    @Test
-    public void testGeneratedType() throws Exception {
-        Map<String, Object> optionsMap = Main.parseArguments("generate", "--data-size", "500", "--data-type", "decimal", "test1.10n");
-        try (IonReader reader = DataGeneratorTest.executeAndRead(optionsMap)) {
-            while (reader.next() != null) {
-                assertSame(reader.getType(), IonType.valueOf(optionsMap.get("--data-type").toString().toUpperCase()));
-            }
-        }
-    }
-
-    /**
-     * Assert the exponent range of generated Ion decimals is conform with the expected range.
-     * @throws Exception if error occurs when executing Ion data generator.
-     */
-    @Test
-    public void testGeneratedDecimalExponentRange() throws Exception {
-        Map<String, Object> optionsMap = Main.parseArguments("generate", "--data-size", "500", "--data-type", "decimal", "--decimal-exponent-range", "[0,10]", "test2.10n");
-        try (IonReader reader = DataGeneratorTest.executeAndRead(optionsMap)) {
-            List<Integer> range = WriteRandomIonValues.parseRange(optionsMap.get("--decimal-exponent-range").toString());
-            while (reader.next() != null) {
-                int exp = reader.decimalValue().scale();
-                assertTrue(exp * (-1) >= range.get(0) && exp * (-1) <= range.get(1));
-            }
-        }
-    }
-
-    /**
-     * Assert the range of coefficient digits number of generated Ion decimals is conform with the expected range.
-     * @throws Exception if error occurs when executing Ion data generator.
-     */
-    @Test
-    public void testGeneratedDecimalCoefficientRange() throws Exception {
```

Comment:
Do we have replacements for these decimal tests?

Reply:
> Do we have replacements for these decimal tests?

Yes, the previous constraints(decimal coefficient digits range and decimal exponent range) have been deprecated. All supported constraints (precision, scale) will be tested by:
```
    @Test
    public void testViolationOfTimestamp() throws Exception {
        DataGeneratorTest.violationDetect(INPUT_ION_TIMESTAMP_FILE_PATH);
    }  
```
I'll discuss with @desaikd to double check whether` ion schema kotlin ` (which is used in the method   DataGeneratorTest.violationDetect ) cover all these constraints violation detection.

Reply:
As per offline discussion,
For `DataGeneratorTest.violationDetect` method we can use `loadSchema()` instead of `newSchema()`. For using `loadSchema` you will need to create a `IonSchemaSystem` and `Authority`. Adding an example here that could be helpful:
```java
IonSchemaSystem ISS = IonSchemaSystemBuilder.standard()
.addAuthority(AuthorityFilesystem(FOLDER_PATH_TO_SCHEMA_FILE))
.build();

Schema schema = ISS.loadSchema(SCHEMA_FILE_NAME);
Type schemaType = schema.getType(SCHEMA_TYPE_NAME);
schemaType.validate(ION_VALUE);
```

`FOLDER_PATH_TO_SCHEMA_FILE` is the root directory path to your schema file.
`SCHEMA_FILE_NAME` is the schema file name (`.isl` file).
`SCHEMA_TYPE_NAME` is the [type](https://amzn.github.io/ion-schema/docs/spec.html#type-definitions) name from schema file. 
`ION_VALUE` is the Ion value to be validated against the schema type.

Reply:
> As per offline discussion, For `DataGeneratorTest.violationDetect` method we can use `loadSchema()` instead of `newSchema()`. For using `loadSchema` you will need to create a `IonSchemaSystem` and `Authority`. Adding an example here that could be helpful:
> 
> ```java
> IonSchemaSystem ISS = IonSchemaSystemBuilder.standard()
> .addAuthority(AuthorityFilesystem(FOLDER_PATH_TO_SCHEMA_FILE))
> .build();
> 
> Schema schema = ISS.loadSchema(SCHEMA_FILE_NAME);
> Type schemaType = schema.getType(SCHEMA_TYPE_NAME);
> schemaType.validate(ION_VALUE);
> ```
> 
> `FOLDER_PATH_TO_SCHEMA_FILE` is the root directory path to your schema file. `SCHEMA_FILE_NAME` is the schema file name (`.isl` file). `SCHEMA_TYPE_NAME` is the [type](https://amzn.github.io/ion-schema/docs/spec.html#type-definitions) name from schema file. `ION_VALUE` is the Ion value to be validated against the schema type.

Thank you, change has been updated.